packages/react-native-codegen/src/parsers/parsers-commons.js

```diff
@@ -92,6 +109,66 @@ function assertGenericTypeAnnotationHasExactlyOneTypeParameter(
   }
 }
 
+function parseObjectProperty(
+  property: NamedShape<Nullable<NativeModuleBaseTypeAnnotation>>,
+  hasteModuleName: string,
+  types: TypeDeclarationMap,
+  aliasMap: {...NativeModuleAliasMap},
+  tryParse: ParserErrorCapturer,
+  cxxOnly: boolean,
+  language: ParserType,
+  translateTypeAnnotation: TranslateTypeAnnotation,
+) {
+  if (
+    property.type !== 'ObjectTypeProperty' &&
+    property.type !== 'TSPropertySignature'
```

Comment:
The Codegen can parse Flow or TypeScript specification files.
We are trying to bring the two together, removing a lot of code duplication.

This check should be exported in a smaller function like:
```ts
function isObjectProperty(property: NamedShape<Nullable<NativeModuleBaseTypeAnnotation>>, language: ParserType): boolean {
  switch language:
    case 'Flow':
      return property.type === 'ObjectTypeProperty';
    case 'TypeScript':
      return property.type === 'TSPropertySignature';
}
```

so that you can use it like this:
```
if (!isObjectProperty(property, language)) {
  // throw the exception
}
```

Reply:
Yeah, that would great way to DRY.
Also, since we expect `boolean` from `isObjectProperty` fn, we should add a `default` case as well.. :)

PS: Thanks for the hints! Will push the changes soon..

Reply:
extracted `objectProperty` check -> `isObjectProperty` fn, export